softoff/softoff.cpp

```diff
@@ -362,26 +367,37 @@ int SoftPowerOff::hostSoftOff(sdeventplus::Event& event)
             exit(-1);
         }
 
-        responseReceived = true;
+        error(
+            "Got the response for set effecter states command, PLDM RC = {RC}",
+            "RC", lg2::hex, static_cast<uint16_t>(response->payload[0]));
+        std::vector<uint8_t> responseMessage;
+        responseMessage.resize(responseMsgSize);
+        memcpy(responseMessage.data(), responseMsg, responseMessage.size());
+        pldm::utils::printBuffer(pldm::utils::Rx, responseMessage);
 
-        // Start Timer
-        using namespace std::chrono;
-        auto timeMicroseconds =
-            duration_cast<microseconds>(seconds(SOFTOFF_TIMEOUT_SECONDS));
+        responseReceived = true;
 
-        auto ret = startTimer(timeMicroseconds);
-        if (ret < 0)
-        {
-            error(
-                "Failure to start Host soft off wait timer, ERRNO = {RET}. Exit the pldm-softpoweroff",
-                "RET", ret);
-            exit(-1);
-        }
-        else
+        if (!noTimeOut)
         {
-            error(
-                "Timer started waiting for host soft off, TIMEOUT_IN_SEC = {TIMEOUT_SEC}",
-                "TIMEOUT_SEC", SOFTOFF_TIMEOUT_SECONDS);
+            // Start Timer
+            using namespace std::chrono;
+            auto timeMicroseconds =
+                duration_cast<microseconds>(seconds(SOFTOFF_TIMEOUT_SECONDS));
```

Comment:
is this really needed to move under the `if` ?

Reply:
yes, we do this only if (!noTimeOut)